File: packages/device_info/lib/src/flutter_config/src/models/src/package_info/package_info.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'package_info.freezed.dart';
part 'package_info.g.dart';

@freezed
class PackageInfo with _$PackageInfo {
  factory PackageInfo(
    final String packageName,
    final String appType,
    final num flags,
    final num category,
    final String? description,
    final String? storageUUID,
    final num uid,
  ) = _PackageInfo;

  factory PackageInfo.fromJson(Map<String, dynamic> json) =>
      _$PackageInfoFromJson(json);
}

File: packages/device_info/android/src/main/kotlin/com/rocket/device/info/models/PackageInfo.kt
package com.rocket.device.info.models

data class PackageInfo(
    val packageName: String,
    val appType: String,
    val flags: Int,
    val category: Int,
    val permissions: String?,
    val storageUUID: String?,
    val uid: Int,
){
    fun toMap(): Map<String, Any> {
        val map = HashMap<String, Any>()
        map["packageName"] = packageName
        map["appType"] = appType
        map["flags"] = flags
        map["category"] = category
        if(permissions != null) map["permissions"] = permissions
        if(storageUUID != null) map["storageUUID"] = storageUUID
        map["uid"] = uid
        return map
    }
}
